.github/workflows/cmake.yml

```diff
@@ -17,45 +17,68 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
-    - name: Create Build Environment
-      # Some projects don't allow in-source building, so create a separate build directory
-      # We'll use this as our working directory for all subsequent commands
-      run: cmake -E make_directory ${{runner.workspace}}/build
+    - uses: actions/cache@v2
+      id: InstalledDependencyCache
+      with:
+        path: ${{runner.workspace}}/install
+        key: InstalledDependencyCache
 
     - name: Install xxHash
```

Comment:
why are we installing these libraries if the cmake is set up to automatically find and downloads them?

Reply:
I'm not 100% sure how to cache the automatically downloaded packages. Without caching, every time we want to run a test it takes a few minutes to just install dependencies, most of the time being taken up by boost. The way I'm doing it now, I'm downloading the dependencies and installing them into a cached directories, so that we don't need to rebuild them every time.

Reply:
then why have a cmake step to install these libraries?

Reply:
For convenience when building from scratch, i guess? I thought it was nice to be able to autodownload everything and it just works out of the box. It's just that for github workflows, since your working environment is reset each time, using autodownload would be wasting time, and I don't know of a good way to cache the results of the autodownload